pandas/core/arrays/datetimelike.py

```diff
@@ -52,6 +52,8 @@
 from pandas.tseries import frequencies
 from pandas.tseries.offsets import DateOffset, Tick
 
+DTScalar = Union[Period, Timestamp, Timedelta, NaTType]
```

Comment:
would put in _typing and call it DatetimeScalar

Reply:
ok

Reply:
Actually looks like we already have DatetimeLikeScalar which just needs to have NaTType added to it

Reply:
updated+green